includes/background-jobs/class-sensei-scheduler-action-scheduler.php

```diff
@@ -59,16 +57,21 @@ public function schedule_job( Sensei_Background_Job_Interface $job, $time = null
 	public function run( Sensei_Background_Job_Interface $job, $completion_callback = null ) {
 		$this->current_job = $job;
 
-		$this->schedule_job( $job );
+		// Schedule action 60 seconds in the future as fallback.
+		$this->schedule_job( $job, time() + 60 );
```

Comment:
If the job delays more than 60 seconds, the same issue could still happen, right? If yes, anyway, it would already be an improvement from what we have currently! =)

Reply:
That's correct. But AFAIK Action Scheduler breaks the execution [before reaching 30s](https://actionscheduler.org/#batch-processing-background-jobs).

But as you suggest yes, this could STILL happen in some weird situations.